File: sphone-api/Domain/Orders/Events/OrderCreatedEvent.cs
using SPhone.Domain.Common;

namespace SPhone.Domain.Orders.Events;

public sealed record OrderCreatedEvent(
    Guid EventId,
    DateTime OccurredAt,
    Guid OrderId,
    Guid CustomerId,
    Guid ProductId,
    decimal MonthlyAmount,
    int TotalMonths) : IDomainEvent
{
    public static OrderCreatedEvent Create(OrderId orderId, Guid customerId, Guid productId, InstallmentPlan plan) =>
        new(Guid.NewGuid(), DateTime.UtcNow, orderId.Value, customerId, productId, plan.MonthlyAmount, plan.TotalMonths);
}

File: sphone-api/src/domain/orders/events/OrderCreatedEvent.ts
import { DomainEvent } from '../../shared/DomainEvent'
import { v4 as uuidv4 } from 'uuid'

export class OrderCreatedEvent implements DomainEvent {
  readonly eventId: string
  readonly occurredAt: Date
  readonly eventType = 'OrderCreated'

  constructor(
    public readonly orderId: string,
    public readonly customerId: string,
    public readonly productId: string,
    public readonly monthlyAmount: number,
    public readonly totalMonths: number,
  ) {
    this.eventId = uuidv4()
    this.occurredAt = new Date()
  }
}
